models/events.js

```diff
@@ -66,8 +68,86 @@ const endActiveEvent = async ({ id, reason, lock }) => {
   }
 };
 
+/**
+ * Stores a reference to a peer in the event document in the Firestore database.
+ * @async
+ * @function
+ * @param {string} peerId - The ID of the peer document to be referenced.
+ * @param {string} eventId - The ID of the event document where the peer reference will be stored.
+ * @throws {Error} If an error occurs while updating the event document.
+ */
+const storePeerReferenceInEvent = async (peerId, eventId) => {
+  const eventDocRef = eventModel.doc(eventId);
+  await eventDocRef.update({
+    peers: Firestore.FieldValue.arrayUnion(peerId),
+  });
+};
+
+/**
+ * Adds a peer to an event in the Firestore database.
+ * @async
+ * @function
+ * @param {Object} peerData - The data of the peer to be added.
+ * @param {string} peerData.name - The name of the peer.
+ * @param {string} peerData.eventId - The unique identifier of the event the peer is being added to.
+ * @param {string} peerData.role - The role of the peer in the event.
+ * @param {Date} peerData.joinedAt - The timestamp indicating when the peer joined the event.
+ * @returns {Promise<Object>} The data of the added peer.
+ * @throws {Error} If an error occurs while adding the peer to the event.
+ */
+const addPeerToEvent = async (peerData) => {
+  try {
+    let docRef;
+    const querySnapshot = await peerModel.where("name", "==", peerData.name).limit(1).get();
```

Comment:
Why are we checking this with name?
Multiple people can have same name

Reply:
I understand we shoudn't be checking with names but from frontend I'm only getting name, eventId, role, joinedAt.

Reply:
This is resolved as per the new requirements.